app/models/user_programme_assessment.rb

```diff
@@ -19,7 +19,7 @@ def enough_activities_for_test?
   end
 
   def num_attempts
-    return 0 if @attempts.nil?
+    return 0 if @attempts.nil? || @attempts&.last&.current_state == StateMachines::AssessmentAttemptStateMachine::STATE_TIMEDOUT.to_s
```

Comment:
I think this could be written differently, as its a lot of `&.`, you might be able to count attempts that are in failed and complete, as the count is still going to include timed out tests.

@attempts could be changed to include only failed and complete, which then fixes the logic of can_take_test_at as well, as it just doesnt include the timed out ones